Firebase/DynamicLinks/Utilities/FDLUtilities.m

```diff
@@ -192,6 +193,23 @@ BOOL FIRDLOSVersionSupported(NSString *_Nullable systemVersion, NSString *minSup
   return timeZoneName;
 }
 
+BOOL FIRDLIsURLForWhiteListedCustomDomain(NSURL *_Nullable URL) {
+  BOOL customDomainMatchFound = false;
+  for (NSURL *allowedCustomDomain in FIRDLCustomDomains) {
+    // All custom domain host names should match at a minimum.
+    if ([allowedCustomDomain.host isEqualToString:URL.host]) {
+      // Next, do a string compare to check if the full path matches as well.
+      if (([URL.absoluteString rangeOfString:allowedCustomDomain.absoluteString
```

Comment:
This will match same level subpaths as well, right? Is that intended? Ex:

```
// allowedCustomDomain
https://google.com/foo

// Matches:
https://google.com/foo
https://google.com/foo2
https://google.com/fooBar
```

Reply:
Chatted offline with Michael. We will consider all domainURIPrefixes for exact matches. PTAL.